### PR TITLE
feat(editor): Add readonly mode to new canvas (no-changelog)

### PR DESCRIPTION
**Pull request by alexgrozav**
## Summary

https://github.com/user-attachments/assets/ed7cc867-822e-4f50-9388-964b188fc9a2

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/N8N-7475/as-a-user-i-want-to-be-able-to-freely-switch-between-executions-and
https://linear.app/n8n/issue/N8N-7483/as-a-user-i-want-the-execution-view-to-be-rendered-using-the-new

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/components/canvas/Canvas.vue [hunk 1]
```diff
@@ -322,6 +331,10 @@ function onContextMenuAction(action: ContextMenuAction, nodeIds: string[]) {
 onMounted(() => {
 	props.eventBus.on('fitView', onFitView);
 	props.eventBus.on('selectNodes', onSelectNodes);
+
+	if (props.readOnly) {
+		setReadonly(props.readOnly);
```
Comment (elsmr) [inline packages/editor-ui/src/components/canvas/Canvas.vue:336]: nit: could avoid this by putting `{ immediate: true }` on the watcher

File: packages/editor-ui/src/views/NodeView.v2.vue [hunk 2]
```diff
@@ -910,8 +911,101 @@ function trackRunWorkflowToNode(node: INodeUi) {
...
+		toast.showError(error, i18n.baseText('nodeView.showError.openExecution.title'));
+		return;
+	}
+	if (data === undefined) {
+		throw new Error(`Execution with id "${executionId}" could not be found!`);
+	}
+
+	await initializeData();
+	await initializeWorkspace(data.workflowData);
+	workflowsStore.setWorkflowExecutionData(data);
```
Comment (elsmr) [inline packages/editor-ui/src/views/NodeView.v2.vue:931]: Why is all this logic in `openExecution` and `onExecutionOpenedWithError` needed? The execution itself is displayed in an iframe so why do we need to call `initializeWorkspace` etc?

In a perfect world™️, should the executions view handle all this stuff? and NodeView just do a router navigation?

**Review by elsmr — DISMISSED** [review]
Great work 😄 2 things I found:

The last node in an execution displays this little badge:

image


Not critical at all but, the context menu button looks a bit lonely in the center 😛 maybe we could right-align it? or show the other buttons in a disabled state.

image

File: packages/editor-ui/src/views/NodeView.v2.vue (hunk 2 above)
Comment (alexgrozav) [reply packages/editor-ui/src/views/NodeView.v2.vue:931]: Technically, if there would be no iframe, we'd be able to do this inside of the executions view. This is basically us hacking our way around it. 

I'm adding a task so we remember to clean this up after we remove the need for an iframe.

**Review by elsmr — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.52.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.52.0)

**Pull request merged**